google/cloud/storage/grpc_plugin_test.cc

```diff
@@ -0,0 +1,129 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/grpc_plugin.h"
+#include "google/cloud/storage/internal/curl_client.h"
+#include "google/cloud/storage/internal/grpc_client.h"
+#include "google/cloud/storage/internal/hybrid_client.h"
+#include "google/cloud/storage/internal/retry_client.h"
+#include "google/cloud/testing_util/scoped_environment.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace storage_experimental {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+using ::google::cloud::storage::internal::ClientImplDetails;
+using ::google::cloud::storage::internal::CurlClient;
+using ::google::cloud::storage::internal::GrpcClient;
+using ::google::cloud::storage::internal::HybridClient;
+using ::google::cloud::storage::internal::RetryClient;
+using ::google::cloud::testing_util::ScopedEnvironment;
+using ::testing::IsNull;
+using ::testing::NotNull;
+
+TEST(GrpcPluginTest, MetadataConfigCreatesGrpc) {
+  // Disable the RestClient, while this is not the default, it is enabled in
+  // some of our CI builds.
```

Comment:
maybe: `// Explicitly disable the RestClient, which may be enabled by our CI builds.` here and elsewhere